tests/jballerina-unit-test/src/test/resources/test-src/types/constant/string_template_constant_negative.bal

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) 2025 WSO2 LLC. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+const v1 = {a:4};
+const v2 = string `hello${v1}`;
+
+boolean bool = true;
+const v3 = string `This is ${bool}`;
+const v4 = string `This is ${foo()}`;
+
+function foo() returns boolean {
+    return false;
+}
```

Comment:
Shall we add negative tests where we refer non constant values in the template expression.

```ballerina
final string foo = "FOO";
string bar = "BAR";
const A = string `${foo}:${bar}`
```